backend/services/office_hours/section.py

```diff
@@ -0,0 +1,787 @@
+__copyright__ = "Copyright 2024"
+__license__ = "MIT"
+
+
+class OfficeHoursSectionService:
+    """Service that performs all of the actions on the `OfficeHoursSection` table"""
+
+    def __init__(
+        self,
+        session: Session = Depends(db_session),
+    ):
+        """Initializes the database session."""
+        self._session = session
+
+    def create(
+        self,
+        subject: User,
+        oh_section: OfficeHoursSectionDraft,
+        academic_ids: list[int],
+    ) -> OfficeHoursSectionDetails:
+        """Creates a new office hours section.
+
+        Args:
+            subject (User): a valid User model representing the currently logged in User
+            oh_section (OfficeHoursSectionDraft): OfficeHoursSection Draft to add to table
+            academic_ids (list[int]): List of academic section IDs to create an OH Section for
+        Returns:
+            OfficeHoursSectionDetails: Object added to table
+        """
+
+        # PERMISSIONS
+
+        # 1. Check If Give Academic Sections IDs Are Valid
+        query = select(SectionEntity).where(SectionEntity.id.in_(academic_ids))
+        academic_section_entities = self._session.scalars(query).all()
+
+        if len(academic_section_entities) != len(academic_ids):
+            raise ResourceNotFoundException(
+                f"Unable to Fetch All Academic Sections. Only {len(academic_section_entities)} out of {len(academic_ids)} was found."
+            )
+
+        # 2. Check if Already Have an Office Hours Event
+        for entity in academic_section_entities:
+            if entity.office_hours_id is not None:
+                raise Exception("Office Hours Section Already Exists!")
+
+        # 3. Check If User is a Member in a Section and Has Proper Role to Create
+        for academic_id in academic_ids:
+            section_member_entity = self._check_membership_by_user_id_section_id(
+                subject.id, academic_id
+            )
+
+            if section_member_entity.member_role != RosterRole.INSTRUCTOR:
+                raise PermissionError(
+                    f"Section Member is not an Instructor. User Does Not Have Permisions Create an Office Hours Section."
+                )
+
+        # Query and Execution to Update All Academic Section With New OH Section ID
+        oh_section_entity = OfficeHoursSectionEntity.from_draft_model(oh_section)
+
+        # Add new object to table and commit changes
+        self._session.add(oh_section_entity)
+
+        # Save Changes To Get OH Section ID
+        self._session.commit()
+
+        # Now Update Office Hours ID
+        for entity in academic_section_entities:
+            entity.office_hours_id = oh_section_entity.id
+
+        self._session.commit()
+
+        # Return details model version of added object
+        return oh_section_entity.to_details_model()
+
+    def get_all_sections(self, subject: User) -> list[OfficeHoursSectionDetails]:
```

Comment:
Subject is not used here, so it can either be removed or it can be used in a permission check, if needed!